### PR TITLE
Nuvoton: M487 support OTA-v3

### PR DESCRIPTION
**Pull request by cyliangtw**
Description
-----------
This PR purpose is not to ask any qualification logo and not plan to process OCW.
It is one normal PR to support \ \amazon-freertos\vendors\vendor\boards\board\ports\ota_pal_for_aws for new OTA
changes from FreeRTOS.

This PR add ports of M487 ota and it could pass aws_demos with `CONFIG_OTA_MQTT_UPDATE_DEMO_ENABLED` & aws_tests including FULL_OTA_AGENT_ENABLED, FULL_OTA_PAL_ENABLED and MQTTv4_ENABLED.

To run aws_tests by cmake, the command line as: `$ cmake -DVENDOR=nuvoton -DBOARD=numaker_iot_m487_wifi -DCOMPILER=arm-keil -S. -Bbuild -G"Unix Makefiles" -DAFR_ENABLE_TESTS=1`

To run aws_demos by cmake, the command line as: `$ cmake -DVENDOR=nuvoton -DBOARD=numaker_iot_m487_wifi -DCOMPILER=arm-keil -S. -Bbuild -G"Unix Makefiles"`

Attached aws_demo & aws_test log files for your reference.
[afr_ota_v3_test.log](https://github.com/aws/amazon-freertos/files/6776133/afr_ota_v3_test.log)
[afr_ota_v3_demo.log](https://github.com/aws/amazon-freertos/files/6776134/afr_ota_v3_demo.log)

About loader part, we have one for customer reference and our chip has H/W secure boot capacity, it really depends on customer requirement. The loader’s major implementation is how to cooperate with OTA-client and secure boot could rely on M487 solution https://www.nuvoton.com/support/technical-support/technical-articles/TSNuvotonTechBlog-000086/.

The above loader part was mentioned  in #2754  for previous OTA support.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

@yngki 
@aggarw13

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by cyliangtw** [comment]
This PR also could pass IDT test and attached 2 IDT test log files as below:

`$ ./devicetester_win_x86-64.exe run-suite --group-id OTACore --userdata userdata.json
`
[OTACore__Full_OTA_PAL.log](https://github.com/aws/amazon-freertos/files/6781214/OTACore__Full_OTA_PAL.log)
[test_manager.log](https://github.com/aws/amazon-freertos/files/6781216/test_manager.log)


```
aseId=Full_OTA_PAL deviceId=M487
time="2021-07-08T11:03:26+08:00" level=info msg=All tests finished. executionId=b5a224b3-df96-11eb-82d4-54e1add44dc7
time="2021-07-08T11:03:27+08:00" level=info msg=

========== Test Summary ==========
Execution Time:         15m58s
Tests Completed:        25
Tests Passed:           25
Tests Failed:           0
Tests Skipped:          0
----------------------------------
Test Groups:
Optional Test Groups:
    OTACore:            PASSED
----------------------------------
Path to Test Execution Logs: C:\AWS\OTA_v3\devicetester_freertos_win_1.5.0\results\b5a224b3-df96-11eb-82d4-54e1add44dc7\logs
Path to Aggregated JUnit Report: C:\AWS\OTA_v3\devicetester_freertos_win_1.5.0\results\b5a224b3-df96-11eb-82d4-54e1add44dc7\FRQ_Report.xml
==================================

```

**Comment by cyliangtw** [comment]
@aws/freertos-pr-bar-raisers, is there any update ?

File: vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/ota_config.h
Comment (pvyawaha) [inline vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/ota_config.h:84]: We have observed this timeout to be too small in slower network , please increase it to atleast 10000 ( 10 secs )

File: vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/ota_config.h
Comment (pvyawaha) [inline vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/ota_config.h:95]: Please set this default to 128

File: vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_tests/config_files/ota_config.h
Comment (pvyawaha) [inline vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_tests/config_files/ota_config.h:85]: Like earlier comment , please set this to atleast 10000

**Review by pvyawaha — CHANGES_REQUESTED** [review]

File: vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/ota_config.h
Comment (cyliangtw) [reply vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/ota_config.h]: @pvyawaha ,  thanks of your suggestion.

File: vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/ota_config.h
Comment (cyliangtw) [reply vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/ota_config.h]: @pvyawaha ,  yes, 128 should be better.

File: vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_tests/config_files/ota_config.h
Comment (cyliangtw) [reply vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_tests/config_files/ota_config.h]: @pvyawaha ,  thanks of your comment. Fixed in commit 01b0e62.

**Comment by cyliangtw** [comment]
@pvyawaha ,  
After applied your suggestion in commit 01b0e62, is there any other concern in your review ?

**Review by pvyawaha — DISMISSED** [review]

**Comment by cyliangtw** [comment]
@aws/freertos-pr-bar-raisers,
Did you finish this PR review ?
If it's, I will update the last commit to sync project files *.uvproj with AFR mainline.

**Review by cobusve — DISMISSED** [review]

**Comment by cyliangtw** [comment]
@pvyawaha , @cobusve ,
Thanks of your review.
After got your approval of this PR, I rebase mainline to sync project files *.uvproj of PR #3370.
And I also verified OK on build code & test OTA.
Please tag approved again.

**Review by paulbartell — APPROVED** [review]

**Pull request merged**
